geode-for-redis/src/main/java/org/apache/geode/redis/internal/data/delta/ReplaceByteArrays.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.apache.geode.redis.internal.data.delta;
+
+import static org.apache.geode.DataSerializer.readByteArray;
+import static org.apache.geode.internal.InternalDataSerializer.readArrayLength;
+import static org.apache.geode.redis.internal.data.delta.DeltaType.REPLACE_BYTE_ARRAYS;
+
+import java.io.DataInput;
+import java.io.DataOutput;
+import java.io.IOException;
+import java.util.Set;
+
+import org.apache.geode.DataSerializer;
+import org.apache.geode.internal.InternalDataSerializer;
+import org.apache.geode.redis.internal.data.AbstractRedisData;
+
+public class ReplaceByteArrays implements DeltaInfo {
+  private final Set<byte[]> byteArrays;
+
+  public ReplaceByteArrays(Set<byte[]> deltas) {
+    this.byteArrays = deltas;
+  }
+
+  public void serializeTo(DataOutput out) throws IOException {
+    DataSerializer.writeEnum(REPLACE_BYTE_ARRAYS, out);
+    InternalDataSerializer.writeArrayLength(byteArrays.size(), out);
+    for (byte[] bytes : byteArrays) {
+      DataSerializer.writeByteArray(bytes, out);
+    }
+  }
+
+  // Create a member set as you read it in
+  public static void deserializeFrom(DataInput in, AbstractRedisData redisData) throws IOException {
+    redisData.applyReplaceByteArraysDelta();
```

Comment:
I think what would be better here is to have applyReplaceByteArraysDelta take a MemberSet. This method will create that MemberSet (call it members) with the size it reads on the next line. Then in the while loop it will just add each byte array it reads to that MemberSet. The last thing it does after the while loop is done is call redisData.applyReplaceByteArraysDelta(members);
All applyReplaceByteArraysDelta does is call persistNoDelta() and this.members = members.
Sorry that this was confusing. I know I talked about doing it this way and the way you did it. I think doing it with a MemberSet will be the most efficient and is consistent with how you did it on the primary.

Reply:
@dschneider-pivotal  I know RedisSortedSets  unionstore and interstore will need to change their code to to store it's changes, because right now it doesn't. It looks like it uses a MemberMap to store the information. Would it be better to keep it generic or would it be better just to have overloading methods one for MemberSet and one for MemberMap.

Reply:
I think RedisSortedSet will have its own DeltaInfo because it has both byte[] and doubles. For an example of this compare AddByteArrays to AddByteArrayDoublePairs. Given that they will have different DeltaInfo classes, I think it makes sense that they would have a different apply*delta method that they call on RedisData. And those different apply*delta methods can then be strongly typed (one with MemberSet and the other with a MemberMap and ScoreSet.